js/preview/src/build.ts

```diff
@@ -43,23 +43,29 @@ export async function make_build({
 							emptyOutDir: true,
 							outDir: join(template_dir, entry),
 							lib: {
-								entry: join(source_dir, entry, "index.ts"),
+								entry: join(source_dir, path),
 								fileName: "index.js",
 								formats: ["es"]
 							},
+							minify: true,
 							rollupOptions: {
 								output: {
-									entryFileNames: "[name].js"
+									entryFileNames: (chunkInfo: PreRenderedChunk) => {
+										if (chunkInfo.isEntry) {
+											return "index.js";
+										}
+										return `${chunkInfo.name.toLocaleLowerCase()}.js`;
+									}
```

Comment:
Okay I've changed this so we always get an `index.js` for the main entry point (which is what we look for when serving the custom component). Nice thing is this will even work if users change their entrypoints in the package.json.